daemon/k8s_watcher.go

```diff
@@ -110,20 +183,49 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 		return nil
 	}
 
-	if err := d.createThirdPartyResources(); err != nil {
-		return fmt.Errorf("Unable to create third party resource: %s", err)
-	}
-
 	restConfig, err := k8s.CreateConfig(d.conf.K8sEndpoint, d.conf.K8sCfgPath)
 	if err != nil {
 		return fmt.Errorf("Unable to create rest configuration: %s", err)
 	}
 
-	tprClient, err := k8s.CreateTPRClient(restConfig)
+	apiextensionsclientset, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("Unable to create rest configuration for k8s CRD: %s", err)
+	}
+
+	clientName := "custom resource definition"
+
+	if err := createCustomResourceDefinitions(apiextensionsclientset); errors.IsNotFound(err) {
+		// If CRD was not found it might mean we are running in k8s <1.7
+		// then we should set up TPR instead
+		log.Debugf("Detected k8s <1.7, using TPR instead of CRD")
```

Comment:
The log says it 'might mean' we are running k8s with version <1.7, but can we be sure of that? We should check the version of K8s via the API if possible, as this log statement might be incorrect based on the above comment.